rules/Php81/NodeFactory/EnumFactory.php

```diff
@@ -91,14 +96,76 @@ private function createEnumCaseFromConst(ClassConst $classConst): EnumCase
         return $enumCase;
     }
 
-    private function createEnumCaseFromDocComment(PhpDocTagNode $phpDocTagNode): EnumCase
+    /**
+     * @param array<string, int|string> $mapping
+     */
+    private function createEnumCaseFromDocComment(PhpDocTagNode $phpDocTagNode, array $mapping = []): EnumCase
     {
         /** @var MethodTagValueNode $nodeValue */
         $nodeValue = $phpDocTagNode->value;
-
+        $enumValue = $mapping[$nodeValue->methodName] ?? $nodeValue->methodName;
         $enumName = strtoupper($nodeValue->methodName);
-        $enumExpr = $this->builderFactory->val($nodeValue->methodName);
+        $enumExpr = $this->builderFactory->val($enumValue);
 
         return new EnumCase($enumName, $enumExpr);
     }
+
+    /**
+     * @return array<string, int|string>
+     */
+    private function generateMappingFromClass(Class_ $class): array
+    {
+        $mapping = [];
+
+        $classMethod = $this->getValuesClassMethod($class);
```

Comment:
You can directly use:

```suggestion
        $classMethod = $class->getMethod('values');
```